packages/context/src/context.ts

```diff
@@ -126,7 +126,7 @@ export class Context {
    * @param key Binding key
    * @returns true if the binding key is found and removed from this context
    */
-  unbind(key: BindingAddress<unknown>): boolean {
+  unbind(key: BindingAddress): boolean {
```

Comment:
Would the following work too?

```ts
unbind<T>(key: BindingAddress<T>): boolean {
```

(Same for other places in this file where a similar change is made.)

Reply:
In your example, the `T` does not help catch type errors as `unbind(bindingAddressWithWrongType` will be happlily accepted by TypeScript. A generic type is only useful if more than one elements of the method signature use it, including params and return value.